types/binary_literal_test.go

```diff
@@ -15,271 +15,269 @@
 package types
 
 import (
-	. "github.com/pingcap/check"
+	"fmt"
+	"testing"
+
 	"github.com/pingcap/tidb/sessionctx/stmtctx"
-	"github.com/pingcap/tidb/util/testleak"
+	"github.com/stretchr/testify/require"
 )
 
-var _ = Suite(&testBinaryLiteralSuite{})
+func TestBinaryLiteral(t *testing.T) {
+	t.Run("TestTrimLeadingZeroBytes", func(t *testing.T) {
+		tbl := []struct {
+			Input    []byte
+			Expected []byte
+		}{
+			{[]byte{}, []byte{}},
+			{[]byte{0x0}, []byte{0x0}},
+			{[]byte{0x1}, []byte{0x1}},
+			{[]byte{0x1, 0x0}, []byte{0x1, 0x0}},
+			{[]byte{0x0, 0x1}, []byte{0x1}},
+			{[]byte{0x0, 0x0, 0x0}, []byte{0x0}},
+			{[]byte{0x1, 0x0, 0x0}, []byte{0x1, 0x0, 0x0}},
+			{[]byte{0x0, 0x1, 0x0, 0x0, 0x1, 0x0, 0x0}, []byte{0x1, 0x0, 0x0, 0x1, 0x0, 0x0}},
+			{[]byte{0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x1, 0x0, 0x0}, []byte{0x1, 0x0, 0x0, 0x1, 0x0, 0x0}},
+		}
+		for _, item := range tbl {
+			b := trimLeadingZeroBytes(item.Input)
+			require.Equal(t, b, item.Expected, fmt.Sprintf("%#v", item))
```

Comment:
For testify, it is `Equal(t, exected, got)`. So all your `require.Equal` got reversed arguments order.

Reply:
Done